java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -124,6 +145,25 @@ open class StorageService : ResurrectorService() {
         ) { "No StoreOptions found in Intent" }
 
         val options = parcelableOptions.actual
+
+        if (0 != applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE) {
+            bindService(devToolsIntent, connection, Context.BIND_AUTO_CREATE)
+            log.debug { "DevToolsBindingContext" }
+
+            return DevToolsBindingContext(
+                stores.computeIfAbsent(options.storageKey) { Store(options) },
+                coroutineContext,
+                stats,
+                devToolsService
+            ) { storageKey, message ->
+                when (message) {
+                    is ProxyMessage.ModelUpdate<*, *, *>,
+                    is ProxyMessage.Operations<*, *, *> -> resurrectClients(storageKey)
+                    is ProxyMessage.SyncRequest<*, *, *> -> Unit
+                }
+            }
+        }
```

Comment:
This is the type of thing that would be good to strip out of this production class all-together using the abstract factory / provider pattern mentioned above with the wrapper approach.